sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreCoordinator.scala

```diff
@@ -168,9 +220,99 @@ private class StateStoreCoordinator(override val rpcEnv: RpcEnv)
         storeIdsToRemove.mkString(", "))
       context.reply(true)
 
+    case SnapshotUploaded(providerId, version, timestamp) =>
+      stateStoreSnapshotVersions.put(providerId, SnapshotUploadEvent(version, timestamp))
+      logDebug(s"Snapshot uploaded at ${providerId} with version ${version}")
+      // Report all stores that are behind in snapshot uploads
+      val (laggingStores, latestSnapshot) = findLaggingStores()
```

Comment:
should we do this on an interval, instead of every time. e.g. in a situation where we have lets say 200 providers and 5 are lagging. Currently every time there is an upload for the 195 other providers, we will log. This may become noisy. Lets do it every x secs or mins to avoid noise?

Reply:
Yeah I agree, there is a lot of noise even when I was just testing. I've added a configuration to only do a lag report on 5 minute intervals, is this a fair interval to start with or should I bump this up to 10 minutes?